contracts/Karmic.sol

```diff
@@ -6,19 +6,56 @@ import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "./Badger.sol";
 
 contract Karmic is Badger{
-    mapping(address => uint256) private boxTokenTiers;
+    mapping(address => BoxToken) private boxTokenTiers;
     uint256 boxTokenCounter;
 
-    constructor(string memory _newBaseUri) Badger(_newBaseUri) {}
+    struct BoxToken{
+        uint256 id;
+        uint256 amount;
+        uint256 funds;
+        bool passedThreshold;
+        uint256 threshold;
+    }
+
+    modifier isBoxToken(address token){
+        require(boxTokenTiers[token].id != 0, "It is not a box token");
+        _;
+    }
+
+
+    constructor(string memory _newBaseUri) Badger(_newBaseUri) {
+        boxTokenCounter = 1;
+        createTokenTier(0, _newBaseUri, false, address(0));
+    }
+
+    fallback() external payable{
```

Comment:
```suggestion
    receive() external payable{
```

I think using a `receive` function would be cleaner, see [docs](https://docs.soliditylang.org/en/v0.8.11/contracts.html?highlight=fallback#fallback-function)